### PR TITLE
Remove useless version checks in REST tests.

### PR DESCRIPTION
**Pull request by jpountz**
Many tests are added with a version check so that they do not run against a
version that doesn't have the feature yet. Master is 7.0, so all tests that
do not run against 6.0+ can be removed and the version check can be removed
on all tests that always run on 6.0+.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/indices.exists_type/10_basic.yml [hunk 1]
```diff
@@ -1,17 +1,11 @@
 ---
 "Exists type":
- - skip:
-    # this will only run in a mixed cluster environment with at least 1 5.x node
-    version: "5.99.99 - "
-    reason:  multiple types are not supported on 6.x indices onwards
```
Comment (bleskes) [inline rest-api-spec/src/main/resources/rest-api-spec/test/indices.exists_type/10_basic.yml]: I think this test can be removed?

Comment (bleskes) [reply rest-api-spec/src/main/resources/rest-api-spec/test/indices.exists_type/10_basic.yml]: Ignore. I see now you removed the extra type

**Review by bleskes — APPROVED** [review]
Thx for the cleanup. Lgtm.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**